src/Microsoft.DotNet.ProjectModel/Files/IncludeFilesResolver.cs

```diff
@@ -74,28 +74,68 @@ public static IEnumerable<IncludeEntry> GetIncludeFiles(IncludeContext context,
                 }
                 else if (isFile && files.Any())
                 {
-                    includeEntries.Add(new IncludeEntry(targetBasePath, files.First()));
+                    includeEntries.Add(
+                        new IncludeEntry(
+                            targetBasePath,
+                            Path.GetFullPath(
+                                Path.Combine(sourceBasePath, PathUtility.GetPathWithDirectorySeparator(files.First().Path)))));
```

Comment:
Consider `ToList`ing the `files` node. You're calling `Count()` \ `Any()` \ and `First()` on it which might result in a lot of work on the backing enumerable